compiler/rustc_resolve/src/imports.rs

```diff
@@ -551,6 +552,33 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     );
                 }
             });
+
+            for (_, resolution) in self.resolutions(module).borrow().iter() {
+                let resolution = resolution.borrow();
+                if let Some(binding) = resolution.binding
+                    && let Some(glob_binding) = resolution.shadowed_glob
+                    && let NameBinding { kind: NameBindingKind::Import { import, .. }, .. } = glob_binding
+                    && let Some(import_id) = import.id()
+                    && let import_def_id = self.local_def_id(import_id)
+                    && self.effective_visibilities.is_exported(import_def_id)
+                    && glob_binding.vis.is_public()
+                    && !binding.vis.is_public()
+                    && let Some(ns) = binding.res().ns()
+                {
+                    self.lint_buffer.buffer_lint_with_diagnostic(
+                        HIDDEN_GLOB_REEXPORTS,
+                        import_id,
+                        binding.span,
+                        "private item shadows glob re-export",
+                        BuiltinLintDiagnostics::HiddenGlobReexports {
+                            name: self.tcx.opt_item_name(import_def_id.to_def_id()).map(|s| s.to_string()),
```

Comment:
```suggestion
                            name: key.ident.name.to_string(),
```
where `key` is taken from `for (key, resolution) in self.resolutions(module)...`